### PR TITLE
feat(bpmn): add compensation props

### LINKED ISSUE #78: Compensation event
* [ ] Wait for completion (default `true`)
* [ ] Activity reference

Cf. https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/lib/provider/bpmn/parts/implementation/CompensateEventDefinition.js

image

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #78
Related to https://github.com/camunda/camunda-modeler/issues/2397

image

image

Note that for `waitForCompletion` the default is `true`.

image

### CODE REVIEW

File: src/provider/bpmn/properties/CompensationProps.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  const businessObject = getBusinessObject(element);
+
+  let parent = businessObject.$parent;
+
+  // (1) get all activities in parent container
+  let activities = getActivitiesForCompensation(parent);
+
+  // (2) if throwing compensation event is inside an EventSubProcess,
+  // also get all activities outside of the event sub process
+  if (is(parent, 'bpmn:SubProcess') && parent.triggeredByEvent) {
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply src/provider/bpmn/properties/CompensationProps.js:228]: Good question. Regarding the [BPMN reference](https://www.omg.org/spec/BPMN/2.0/PDF) (chapter 10.6.1)

> A Compensation Event Sub-Process can recursively trigger compensation for Activities contained in its parent.

The question here is whether this also counts for parents of parents - and whether such a scenario is actually allowed by any process engine (theoretically and practically). [We handled it the same in the old properties panel](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/lib/provider/bpmn/parts/implementation/CompensateEventDefinition.js#L63) so I assumed, it doesn't work recursively.

File: src/provider/bpmn/properties/CompensationProps.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+ * b) a SubProcess, when it is not event based
+ * c) any other Activity, when it is attaching a CompensateEvent of the parent container
+ *
+ * @param {ModdleElement} activity
+ * @param {Array<ModdleElement>} boundaryEvents
+ * @returns {Boolean}
+ */
+function canBeCompensated(activity, boundaryEvents) {
+  return is(activity, 'bpmn:CallActivity') ||
+    (is(activity, 'bpmn:SubProcess') && !activity.triggeredByEvent) ||
```
Comment (pinussilvestrus) [inline src/provider/bpmn/properties/CompensationProps.js:199]: Cf. https://github.com/camunda/camunda-modeler/issues/2397

We also have to check for `!activity.isForCompensation`. Will add it accordingly.

File: src/provider/bpmn/properties/CompensationProps.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxTru) [reply src/provider/bpmn/properties/CompensationProps.js:228]: Well, then let's probably stay with the old implementation (as it seems, nobody complained :-) )

Maybe it makes sense to give the engine team just a quick heads up?

File: src/provider/bpmn/properties/CompensationProps.js (hunk 2 above)
Comment (pinussilvestrus) [reply src/provider/bpmn/properties/CompensationProps.js]: Added accordingly: 7e5dc672f5efb96b9524e494255b3334e4ff3480

**Comment by pinussilvestrus** [comment]
With 7e5dc672f5efb96b9524e494255b3334e4ff3480, ready for review again.

File: src/provider/bpmn/properties/CompensationProps.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply src/provider/bpmn/properties/CompensationProps.js:228]: Makes sense, let's do it 👍

[the remark this answers is not in the document]
Comment (MaxTru) [reply src/provider/bpmn/properties/CompensationProps.js:228]: Will you do that? (giving the heads-up?) 

If not, please let me know and I can send this conversation to them for information :-)

**Review by MaxTru — APPROVED** [review]
Good job!

File: src/provider/bpmn/properties/CompensationProps.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply src/provider/bpmn/properties/CompensationProps.js:228]: Feel free to do it 👍

**Pull request merged**